### PR TITLE
DOC: Clarify dispatch behavior of read_sql

### LINKED ISSUE #18861: DOC: clarify read_sql docstring on relation with other read_sql_ functions
We have 3 (!) functions for reading data from SQL:

`read_sql`
`read_sql_table`
`read_sql_query`

Definitely one of these can be removed.  Probably aiming at `read_sql_query`, since its functionality seems most similar to that of `read_sql`.

`read_sql_table` seems more like a convenience function in place of `SELECT * FROM <table>`, though one could argue that we could accept a table-name in `read_sql` to do the same thing.

### PR DESCRIPTION
**Pull request by gfyoung**
Wasn't particularly clear or prominent in the docs.

Closes #18861.

### CODE REVIEW

File: pandas/io/sql.py [hunk 1]
```diff
@@ -337,15 +337,24 @@ def read_sql(sql, con, index_col=None, coerce_float=True, params=None,
...
+    database table name will be routed to ``read_sql_table``. Note that
+    the delegated function might have more specific notes about their
+    functionality not listed here.
+
     Parameters
     ----------
     sql : string or SQLAlchemy Selectable (select or text object)
-        SQL query to be executed.
-    con : SQLAlchemy connectable(engine/connection) or database string URI
+        SQL query to be executed or a table name from which to select rows.
```
Comment (jorisvandenbossche) [inline pandas/io/sql.py]: I would leave out "from which to select rows" (there is no way to do this)

File: pandas/io/sql.py [hunk 2]
```diff
@@ -337,15 +337,24 @@ def read_sql(sql, con, index_col=None, coerce_float=True, params=None,
     """
     Read SQL query or database table into a DataFrame.
 
+    This function is a convenience wrapper around ``read_sql_table`` and
+    ``read_sql_query`` (for backward compatibility). It will delegate
+    to the specific function depending on the provided input.
+
+    For example, a SQL query will be routed to ``read_sql_query``, while a
```
Comment (jorisvandenbossche) [inline pandas/io/sql.py]: I would leave out "For example", as there are no other options.

(I would also keep those two paragraphs as a single paragraph, as it is explaining one topic.

Comment (gfyoung) [reply pandas/io/sql.py]: Done.

File: pandas/io/sql.py (hunk 1 above)
Comment (gfyoung) [reply pandas/io/sql.py]: Done.

**Comment by gfyoung** [comment]
@jorisvandenbossche: comments addressed.  PTAL.

**Comment by jreback** [comment]
thanks!

**Pull request merged**